vendor/assets/javascripts/better_colorpicker/better_colorpicker.coffee

```diff
@@ -0,0 +1,274 @@
+      <div class="v line"></div>
+    </div>
+    <div class="h-picker">
+      <div class="gradient"></div>
+      <div class="h sline"></div>
+      <div class="h line"></div>
+    </div>
+    <div class="preview"></div>
+    <div class="mbutton msubmit">Save</div>
+    <div class="mbutton mcancel">Exit</div>
+  </div>'''
+
+EDIT_NONE   = 0
+EDIT_HUE    = 1
+EDIT_SATVAL = 2
+EDIT_OPEN = 3
+
+################################################################################
+# Helper function - fill in the rest of the arguments                          #
+################################################################################
+
+addDefaults = (args, defaults) ->
+  if args == undefined
+    args = {}
+  for x of defaults
+    unless x of args
+      args[x] = defaults[x]
+  args
+
+################################################################################
+# JQuery tie-in                                                                #
+################################################################################
+
+$.fn.extend
+  colorpicker: (args) ->
+    args = addDefaults args,
+      onOpen: ->
+      onChange: (val) ->
+      onSubmit: (val) ->
+      onClose: (val) ->
+      hPosition: (w, h) ->
+        100
+      vPosition: (w, h) ->
+        100
+      anchor: $('body')
+      autoClose: true
+      outputType: 'rgb'
+
+    args.this = @
+
+    colorPicker = $(document).data 'better_colorpicker'
+    if colorPicker == undefined
+      colorPicker = new ColorPicker()
+      $(document).data 'better_colorpicker', colorPicker
+
+    colorRef = new ColorPickerRef args, colorPicker
+    colorRef
+
+################################################################################
+# The actual ColorPicker UI element.                                           #
+# Only one of these ever exists on the page.  To interface with the            #
+#   ColorPicker, use ColorPickerRef.                                           #
+################################################################################
+
+class ColorPicker
+  color: {h: 0, s: 1, v: 1}
+  loadColor: {h: 0, s: 1, v: 1}
+  dragMode: EDIT_NONE
+  jqObj: null
+  anchor: null
+
+  constructor: ->
+    @jqObj = $(cpBody)
+
+  # methods
+  openMenu: (x, y, color, anchor, autoClose, @args) =>
+    pAnchor = if @anchor == null then null else @anchor[0]
+    if anchor[0] isnt pAnchor
+      @anchor = anchor
+      @anchor.css
+        position: 'relative'
+      @jqObj.remove()
+      anchor.append @jqObj
+
+    tempColor = @inputColor color
+    if tempColor.isValid()
+      @color = tempColor.toHsv()
+    else
+      @color = {h: 0, s: 1, v: 1}
+    
+    @args.onChange @outputColor @color
+
+    @loadColor = {h: @color.h, s: @color.s, v: @color.v}
+
+    @jqObj.show()
+    @jqObj.css
+      top: x
+      left: y
+
+    @dragMode = EDIT_OPEN
+
+    $(document).on 'mousemove.better_colorpicker', @dragChange
+    $(document).on 'mouseup.better_colorpicker', (e) =>
```

Comment:
mouseup will not work on mobile. http://stackoverflow.com/questions/25135736/does-jquery-mouseup-event-work-on-touch-devices